### PR TITLE
packages: add upstream kata-agent

### PR DESCRIPTION
**Pull request by msanft**
This adds the kata-agent from the Kata upstream sources to our package set. (`kata.kata-agent`) Additionally, it moves the Microsoft kata-agent to `microsoft.kata-agent` (in the new directory structure).

### CODE REVIEW

File: packages/by-name/kata-image/package.nix [hunk 1]
```diff
@@ -5,11 +5,12 @@
 , stdenv
 , stdenvNoCC
 , distro ? "cbl-mariner"
+, microsoft
 , bubblewrap
 , fakeroot
 , fetchFromGitHub
 , fetchurl
-, kata-agent
+, kata-agent ? microsoft.kata-agent
```
Comment (katexochen) [inline packages/by-name/kata-image/package.nix:13]: This doesn't really make sense, as it builds msft's kata-image and you can't just inject the upstream agent. Same for the kernel-uvm.

**Review by katexochen — APPROVED** [review]
Otherwise lgtm, thanks. :seal:

File: packages/by-name/kata-image/package.nix (hunk 1 above)
Comment (msanft) [reply packages/by-name/kata-image/package.nix:13]: What? This imports the microsoft agent? That's what it did before, right?

Comment (katexochen) [reply packages/by-name/kata-image/package.nix:13]: It makes the kata-agent configurable, and let's it default to the microsoft.kata-agent. Instead, we can just always use microsoft.kata-agent (or just kata-agent, when it's in the same subset?).

**Pull request merged**